agent/consul/leader_connect_test.go

```diff
@@ -565,8 +555,8 @@ func TestCAManager_RenewIntermediate_Secondary(t *testing.T) {
 
 	_, activeRoot, err = store.CARootActive(nil)
 	require.NoError(err)
-	require.Equal(intermediatePEM, s2.caManager.getLeafSigningCertFromRoot(activeRoot))
 	require.Equal(connect.HexString(intermediateCert.SubjectKeyId), activeRoot.SigningKeyID)
+	require.Equal(intermediatePEM, s2.caManager.getLeafSigningCertFromRoot(activeRoot))
```

Comment:
I flipped these because when the test fails it is a lot easier for humans to compare two relative short hex values, than a multi-line PEM encoded certificate.

The `t.Log` I added above in these tests allows us to easily see if the problem is that it was not rotated, or something else.